nixos/modules/services/web-servers/meguca.nix

```diff
@@ -28,68 +28,122 @@ in with lib; {
-      description = "Postgresql connection arguments.";
+      example = "postgres://meguca:meguca@localhost:5432/meguca_test?sslmode=disable";
+
+      description = ''
+        Database URL to use during tests.
+
+        This URL only serves as the base. The actual databases are created
+        and dropped during testing automatically with prefixes for each
+        database according to the submodule using this test database for
+        running unit tests. This division allows database-related for each
+        submodule to be run concurrently, overall reducing the runtime of
+        unit tests.
+
+        To allow database creation during tests the role used must have the
+        necessary PostgreSQL permissions. These can be granted by running
+
+          ALTER USER $user_name WITH CREATEDB;
+
+        as the administrator postgres user in the psql shell.
+      '';
     };
 
-    postgresArgsFile = mkOption {
-      type = types.path;
-      default = "/run/keys/meguca-postgres-args";
-      example = "/home/okina/meguca/keys/postgres";
-      description = "Postgresql connection arguments file.";
+    cacheSize = mkOption {
+      type = types.float;
+      default = 128.0;
+      example = 255.9;
+
+      description = ''
+        Size limit of internal cache in MB. Once limit is exceeded, the least
+		    recently used records from the cache will be evicted.
+      '';
     };
 
-    compressTraffic = mkOption {
-      type = types.bool;
-      default = false;
-      description = "Compress all traffic with gzip.";
+    listenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1:8000";
+      example = "127.0.0.1:9001";
+      description = "Address to listen on for incoming connections.";
     };
 
-    assumeReverseProxy = mkOption {
+    reverseProxy = mkOption {
       type = types.bool;
       default = false;
-      description = "Assume the server is behind a reverse proxy, when resolving client IPs.";
+
+      description = ''
+        The server can only be accessed by clients through a reverse proxy
+        like NGINX and thus can safely honour "X-Forwarded-For" headers
+        for client IP resolution.
+      '';
     };
 
-    httpsOnly = mkOption {
-      type = types.bool;
-      default = false;
-      description = "Serve and listen only through HTTPS.";
+    serverConfig = mkOption {
+      type = types.str;
+
+      example = ''
+        {
+          "debug": true,
+          "imager_mode": 0,
+          "database": "postgres://meguca:meguca@localhost:5432/meguca?sslmode=disable",
+          "cache_size": 128.0,
+          "server": {
+            "address": "127.0.0.1:8000",
+            "reverse_proxied": false
+          },
+          "test": {
+            "database": "postgres://meguca:meguca@localhost:5432/meguca_test?sslmode=disable"
+          }
+        }
+      '';
+
+      description = "Meguca server configuration in JSON format.";
+    };
+
+    serverConfigFile = mkOption {
```

Comment:
it is redundant with serverConfig, people can use `serverConfig = builtins.readFile`instead.